src/librustc_mir/transform/const_prop.rs

```diff
@@ -370,6 +371,12 @@ impl<'a, 'mir, 'tcx> ConstPropagator<'a, 'mir, 'tcx> {
             Rvalue::Discriminant(..) => None,
 
             Rvalue::Cast(kind, ref operand, _) => {
+                // Don't allow casts to function pointers because the pointers point into CTFE
```

Comment:
that shouldn't be a problem, the collector will figure this out. We already have this stuation if you create a function pointer in a constant and use that constant to initialize a variable.